src/gallium/drivers/lima/lima_draw.c

```diff
@@ -580,6 +580,15 @@ lima_pack_render_state(struct lima_context *ctx)
          PIPE_BLENDFACTOR_ONE, PIPE_BLENDFACTOR_ZERO);
    }
 
+   if (rt->colormask & PIPE_MASK_R)
```

Comment:
For optimization: you could use one shift for all four bits here, since the defines match the bits you want to set.
```
render->alpha_blend |= (rt->colormask & PIPE_MASK_RGBA) << 28;

```